docs/logql.md

```diff
@@ -0,0 +1,159 @@
+
+The query language is still under development to support more features, for
+example:
+
+- `AND` / `NOT` operators
+- JSON accessors for filtering of JSON-structured logs
+- Context (like `grep -C n`)
+
+## Counting logs
+
+LogQL also supports functions that wrap a query and allow for counting entries
+per stream.
+
+### Range Vector aggregation
+
+LogQL shares the same [range vector](https://prometheus.io/docs/prometheus/latest/querying/basics/#range-vector-selectors)
+concept from Prometheus, except the selected range of samples include a value of
+1 for each log entry. An aggregation can be applied over the selected range to
+transform it into an instance vector.
+
+`rate` calculates the number of entries per second while `count_over_time`
+counts the entries for each log stream within the range.
+
+> `count_over_time({job="mysql"}[5m])`
+
+This example counts all the log lines within the last five minutes for the
+MySQL job.
+
+> `rate( ( {job="mysql"} |= "error" != "timeout)[10s] ) )`
+
+This example demonstrates that a fully LogQL query can be wrapped in the
+aggregation syntax, including filter expressions. This example gets the
+per-second rate of all non-timeout errors within the last ten seconds for the
+MySQL job.
+
+### Aggregation operators
+
+Like [PromQL](https://prometheus.io/docs/prometheus/latest/querying/operators/#aggregation-operators),
+LogQL supports a subset of built-in aggregation operators that can be used to
+aggregate the element of a single vector, resulting in a new vector of fewer
+elements but with aggregated values:
+
+- `sum`: Calculate sum over labels
+- `min`: Select minimum over labels
+- `max`: Select maximum over labels
+- `avg`: Calculate the average over labels
+- `stddev`: Calculate the population standard deviation over labels
+- `stdvar`: Calculate the population standard variance over labels
+- `count`: Count number of elements in the vector
+- `bottomk`: Select smallest k elements by sample value
+- `topk`: Select largest k elements by sample value
+
+The aggregation operators can either be used to aggregate over all label
+values or a set of distinct label values by including a _without_ or a
+_by_ clause:
+
+> `<aggr-op>([parameter,] <vector expression>) [without|by (<label list>)]`
+
+`parameter` is only required when using `topk` and `bottomk`. `topk` and
+`bottomk` are different from other aggregators in that a subset of the input
+samples, including the original labels, are returned in the result vector. `by`
+and `without` are only used to group the input vector.
+
+The `without` cause removes the listed labels from the resulting vector, keeping
+all others. The `by` clause does the opposite, dropping labels that are not
+listed in the clause, even if their label values are identical between all
+elements of the vector.
+
+#### Examples
+
+> `topk(10,sum(rate({region="us-east1"}[5m]) by (name))`
+
+Gets the top 10 applications by the highest log throughput.
```

Comment:
In the rest of the page we explain the example first and then we show it. Here it's the inverse. We may keep the same pattern, so:

```
Gets the top 10 applications by the highest log throughput:

`topk(10,sum(rate({region="us-east1"}[5m]) by (name))`
```